### ISSUE TITLE
Remove functions that do not block the event loop as much as possible

### ISSUE THREAD

**Issue by zFernand0** [zowe/zowe-explorer-vscode#611]
Looking at the code around and thinking a bit more about this, I believe we should stick to a single paradigm - unless there is a reason to do things otherwise. All of our code uses async/await so I believe it would be better to use it here as well.
```suggestion
import {promisify} from 'util';
const exists = util.promisify(fs.exists);
        if (await exists(beforeFullPath)) {
```
and similarly for `fs.unlink` bellow

_Originally posted by @VitGottwald in https://github.com/zowe/vscode-extension-for-zowe/pull/603#discussion_r390818670

**Comment by VitGottwald** [comment]
Thank you @stepanzharychevbroadcom for pointing out we use synchronous node methods many places and thanks @zFernand0 for moving this to a separate issue.

Does anyone from the team know the background how we ended up using fs.*sync method? Was it just by convenience or is there a rationale for using them?

**Comment by VitGottwald** [comment]
The reason I am pointing this out is that `sync` methods go agains the design principles of nodejs because they block the event loop.

To my knowledge they were unwillingly added only after people complained about "callback hell" in simple scripts. Now when we have async/await we can easily express the same logic without blocking the event loop and without having a callback hell.

On the other hand if our code relies on blocking the event loop I believe it would deserve extra attention and documentation - perhaps in the form of a separate module where we gather such utility functions for clarity (like the combination of fs.exists and fs.unlik) for example.

**Comment by zFernand0** [comment]
This enhancement has had no activity for 12 months. The issue also has less than 5 up-votes by the community. No action on this enhancement is targeted for the next 2 calendar quarters. Therefore, this enhancement is being closed. If you feel that this enhancement should continue to be available for community up-votes, you may reopen this issue.
